src/term/transform/encode_pattern_matching.rs

```diff
@@ -180,6 +193,61 @@ fn make_leaf_pattern_matching_case(
   add_case_to_book(book, crnt_name.clone(), term);
 }
 
+fn make_num_pattern_matching_case(
+  book: &mut Book,
+  def_type: &[Type],
+  def_id: DefId,
+  crnt_name: &Name,
+  crnt_rules: Vec<usize>,
+  match_path: Vec<Pattern>,
+) {
+  use MatchNum::*;
+
+  fn filter_rules(def_rules: &[Rule], crnt_rules: &[usize], arg_idx: usize, Name(ctr): &Name) -> Vec<usize> {
+    crnt_rules
+      .iter()
+      .copied()
+      .filter(|&rule_idx| match &def_rules[rule_idx].pats[arg_idx] {
+        Pattern::Var(_) => false,
+        Pattern::Ctr(_nam, _) => false,
+        Pattern::Num(Succ(_)) => ctr == "+",
```

Comment:
Is there any case where the constructor name will not be `"+"`? If not, we can just have this return true always

Reply:
Isn't it better to leave the check in case it is different somehow, to help catch errors when maintaining the code?

Reply:
It won't catch the error, it will misteriously generate incorrect compilations because this expression will return `false`.

These names are created 10 lines below
`for (next_ctr, next_ctr_args) in vec![(Name::new("0"), vec![]), (Name::new("+"), vec![Name::new("p")])] {`, and they're not really relevant for this function. They're only used to generate the correct name for the next subfunctions.

Reply:
I see